### PR TITLE
add rule to manage secrets in the kubeapps namespace

### PR DESCRIPTION
**Pull request by sameersbn**
The dashboard creates a Secret resource if a chart repo requires authentication. This 
PR adds a rule to the `kubeapps-kube-api` role to be able to do that.

Additionally some minor fixes have been imported from bitnami-labs/helm-crd/pull/23

### CODE REVIEW

File: cmd/chart-repo/utils.go [hunk 1]
```diff
@@ -41,8 +41,9 @@ import (
 )
 
 const (
-	chartCollection      = "charts"
-	chartFilesCollection = "files"
+	chartCollection       = "charts"
+	chartFilesCollection  = "files"
+	defaultTimeoutSeconds = 180
```
Comment (prydonius) [inline cmd/chart-repo/utils.go]: 180 is way too high here, we don't want to wait 2 minutes if an icon takes too long to fetch, it's better to skip it and process it next time. I suggest 30s, or keep it at 10s. This is not the same case as helm-crd, as this command runs every hour.

File: manifests/kube-api.jsonnet [hunk 2]
```diff
@@ -13,6 +13,12 @@ local kube = import "kube.libsonnet";
         resources: ["configmaps"],
         verbs: ["get", "list"],
       },
+      // Kubeapps creates Secrets with authorization token data for private chart repos
+      {
+        apiGroups: [""],
+        resources: ["secrets"],
+        verbs: ["get", "list", "create", "delete"],
```
Comment (prydonius) [inline manifests/kube-api.jsonnet]: AFAIK we don't need delete because we use GC. Do we need get/list? Do we display the secret?

Comment (sameersbn) [reply manifests/kube-api.jsonnet]: we don't use the list and delete, but the Secret api has the methods implemented

File: cmd/chart-repo/utils.go (hunk 1 above)
Comment (sameersbn) [reply cmd/chart-repo/utils.go]: fair enough

**Comment by sameersbn** [comment]
updated!

File: cmd/chart-repo/utils.go (hunk 1 above)
Comment (sameersbn) [reply cmd/chart-repo/utils.go]: reverted to 10s

File: manifests/kube-api.jsonnet (hunk 2 above)
Comment (sameersbn) [reply manifests/kube-api.jsonnet]: dropped all but `create` as thats the one we need

Comment (prydonius) [reply manifests/kube-api.jsonnet]: > but the Secret api has the methods implemented
of course, but the point is to restrict any operations that aren't required. This way, Kubeapps doesn't have access to secrets it shouldn't have (technically Kubeapps already has cluster-wide access to read secrets, but this is not as important with the auth changes).

**Comment by prydonius** [comment]
thanks for fixing this!

**Pull request merged**